List commands hotkey

hanspr opened the pull request:
In terminal list hotkeys configured in remote commands

hanspr committed:
do not close socket on disconnect only at the time of a full stop

hanspr committed:
merged

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
if string has utf8 characters will copy incomplete text because should count bytes

hanspr committed:
remove commented tests

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
ignore vscode dir

hanspr committed:
merge

hanspr committed:
Merge branch 'loki' of https://github.com/asbru-cm/asbru-cm into loki

hanspr committed:
ignore local ide

hanspr committed:
list hotkeys on terminal

hanspr committed:
Merge branch 'loki' into listhk

gfrenoy commented:
@hanspr I cannot make this to work...

If the hotkey has never been used before, nothing is displayed.
If I first use the hotkey, then I see it in the list but there is no label.

I'm not 100% sure but I believe the way we loop into the $kbs variable is not correct.

I'll give another look tomorrow.

hanspr commented:
That is shown up on using it first it is a weird behavior, I will try to replicate during the week.

Regarding the missing label, I assume that there is a description in your configuration of the remote command (I ask because have seen from previous posts from users that many people tend to leave this blank), and the label comes from the description.

<img width="852" height="124" alt="imagen" src="https://github.com/user-attachments/assets/01c2d8d2-d24c-4c79-bf71-eec13d0b48fe" />

gfrenoy commented:
> I assume that there is a description in your configuration

Yes, there is one.

Will try again tomorrow and give more details.

hanspr commented:
I started asbru with a new clean configuration: `./asbru-cm --config-dir=/home/hans/.config/asbru-new --verbose`

Created a new remote command

<img width="857" height="147" alt="imagen" src="https://github.com/user-attachments/assets/685a3897-5d17-4948-9bf6-8edd89874a69" />

Opened a local terminal, and tested, was shown correctly without using it, or anything

<img width="513" height="100" alt="imagen" src="https://github.com/user-attachments/assets/fa14fdf8-66d0-412d-b1e3-431f0baaf653" />

Closed the terminal, Added a second remote command

<img width="860" height="279" alt="imagen" src="https://github.com/user-attachments/assets/760f1f10-6c2f-4582-bc7d-16fd68b8fb1c" />

Opened a new local terminal. and tested

<img width="472" height="104" alt="imagen" src="https://github.com/user-attachments/assets/56906bb9-c512-435c-a67d-1b3670583460" />

---

So, removed the clean configuration and recreated again from scratch, and this time I opened the terminal first. Then created the hotkey for the remote command, and then checked if it was shown, and it worked too.

<img width="846" height="177" alt="imagen" src="https://github.com/user-attachments/assets/37b20fda-a45b-48c1-a390-0f11b31af5e1" />

<img width="474" height="77" alt="imagen" src="https://github.com/user-attachments/assets/036e8927-59dd-4585-a7d2-14c5ac644bb6" />

So at least, from a new configuration it worked, so it means the problem should be around pre existing configurations. I will test that next.

gfrenoy commented:
Ok I see the « problem » !  I tested this with **connection** remote commands.  Not the global ones !

So for the global commands, it works fine.  But if you define a command at the connection level, it behaves like described above.  I guess this is because there is one level more in the structure.

Do we want to support the connection commands as well ?

gfrenoy commented:
This is how the data looks like when you create connection-specific remote (or local) commands:

```
{
          '9b42d0c7-57c1-46dd-9093-43f857a99d0f' => {
                                                      'terminal' => {
                                                                      'Ctrl+l' => [
                                                                                    'User hotkey',
                                                                                    'HOTKEY_CMD:List Files Remote',
                                                                                    'ls -l
'
                                                                                  ],
                                                                      'AltCtrl+l' => [
                                                                                       'User hotkey',
                                                                                       'HOTKEY_CMD:List Files Local',
                                                                                       'ls -l
'
                                                                                     ]
                                                                    }
                                                    },
          'terminal' => {
                          'AltCtrl+k' => [
                                           'User hotkey',
                                           'HOTKEY_CMD:List Files Remote Global',
                                           'ls -l
'
                                         ]
                        }
        };
```

The first part is ignored.

gfrenoy commented:
Made a few minor changes to handle the various cases.  It works fine, thanks a lot for this cool improvement :)
